### PR TITLE
Goated Experimental changes

### PR DESCRIPTION
**Pull request by Spyrxo**
### quantum ssm-multiverse code changes

- Integrated Maven into the file-code system, to handle libraries, setup, shading and compiling.
- Refactored the codebase to correctly implement git maven hierarchy.
- Cleaned the readme file with proper instructions and information.
- Fixed inconsistencies and errors which existed due to the original project using Java 14 instead of Java 8.
- Minor tweaks and code fixes (mainly getting rid of the copy-pasted spahetti if else statements)

### CODE REVIEW

File: src/main/java/xyz/whoneedspacee/ssmos/managers/DisguiseManager.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public class DisguiseManager implements Listener, Runnable {
     // Player Joining
     // Player Leaving
     // Player Respawning (when a player dies we need to reload all disguises for them)
-    // Everything else should be handled by kits equipping disguises
+    // Everything else should be handled by whoneedspacee.ssmos.kits equipping disguises
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/managers/DisguiseManager.java]: What

File: src/main/resources/plugin.yml [hunk 2]
```diff
@@ -34,7 +35,7 @@ commands:
     usage: /spectate
     aliases: [spec]
   setplaying:
-    description: 'Forces a kits abilities to be active.'
+    description: 'Forces a whoneedspacee.ssmos.kits whoneedspacee.ssmos.abilities to be active.'
```
Comment (Whoneedspacee) [inline src/main/resources/plugin.yml]: Something about this feels wrong

File: src/main/java/xyz/whoneedspacee/ssmos/Main.java [hunk 3]
```diff
@@ -145,7 +147,8 @@ public void onDisable() {
         for (Disguise disguise : DisguiseManager.disguises.values()) {
             disguise.deleteLiving();
         }
-        List<SmashServer> to_delete = List.copyOf(GameManager.servers);
+        //List<SmashServer> to_delete = List.copyOf(GameManager.servers);       // gholemFix: replaced because List.copyOf doesn't exist in Java 8
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/Main.java]: You don't need to comment old fixes

File: src/main/java/xyz/whoneedspacee/ssmos/abilities/Ability.java [hunk 4]
```diff
@@ -1,6 +1,6 @@
-package ssm.abilities;
+package xyz.whoneedspacee.ssmos.abilities;
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/abilities/Ability.java:1]: Why xyz for package naming? Not a jab or anything just genuinely curious.

File: src/main/java/xyz/whoneedspacee/ssmos/abilities/boss/WhirlpoolShotgun.java [hunk 5]
```diff
@@ -17,7 +18,7 @@ public WhirlpoolShotgun() {
         super();
         this.name = "Whirlpool Shotgun";
         this.cooldownTime = 5;
-        this.usage = AbilityUsage.RIGHT_CLICK;
+        this.usage = Attribute.AbilityUsage.RIGHT_CLICK;
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/abilities/boss/WhirlpoolShotgun.java:21]: Is there a need to hard specify Attribute.AbilityUsage?

File: src/main/java/xyz/whoneedspacee/ssmos/kits/Kit.java [hunk 6]
```diff
@@ -91,7 +90,7 @@ public void updatePlaying(short new_state, boolean reload_hotbar) {
             return;
         }
         boolean game_hotbar = GameState.isStarting(new_state) || GameState.isPlaying(new_state) || new_state == GameState.GAME_ENDING;
-        // Set hotbar and register or unregister events for attributes
+        // Set hotbar and register or unregister whoneedspacee.ssmos.events for whoneedspacee.ssmos.attributes
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/kits/Kit.java]: Weird comment here

File: src/main/java/xyz/whoneedspacee/ssmos/kits/Kit.java [hunk 7]
```diff
@@ -15,18 +16,16 @@
...
+import xyz.whoneedspacee.ssmos.managers.smashserver.SmashServer;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
 public abstract class Kit implements Listener {
 
-    /*  You must add your new kits to the allkits list in the main SSM.java file! */
+    /*  You must add your new whoneedspacee.ssmos.kits to the allkits list in the main SSM.java file! */
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/kits/Kit.java]: Another weird comment

**Review by Whoneedspacee — COMMENTED** [review]
I am really not a fan of the massive amount of file changes you've done here, I really do not feel like reviewing 500 files, in the future please split up your pull requests to be smaller. I'll need to think about this.

File: src/main/java/xyz/whoneedspacee/ssmos/managers/disguises/Disguise.java [hunk 8]
```diff
@@ -338,7 +388,15 @@ public float getPitch() {
     }
 
     public void playDamageSound() {
-        owner.getWorld().playSound(owner.getLocation(), getDamageSound(), getVolume(), getPitch());
+        // gholemFix: similar to the damageUtil sound method fix, if its guardian, use NMS make sound with the string, else use bukkit api
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/managers/disguises/Disguise.java]: This should have been handled in the guardian disguise class, as of right now I don't think this even does anything?

**Comment by Whoneedspacee** [comment]
I appreciate the work you're trying to do but this is too much for me to review, if we can talk about it further I'd be possibly open to merging it.

File: src/main/java/xyz/whoneedspacee/ssmos/managers/DisguiseManager.java (hunk 1 above)
Comment (Spyrxo) [reply src/main/java/xyz/whoneedspacee/ssmos/managers/DisguiseManager.java]: intellij auto-refactor outcome...

File: src/main/resources/plugin.yml (hunk 2 above)
Comment (Spyrxo) [reply src/main/resources/plugin.yml]: intellij auto-refactor outcome...

File: src/main/java/xyz/whoneedspacee/ssmos/Main.java (hunk 3 above)
Comment (Spyrxo) [reply src/main/java/xyz/whoneedspacee/ssmos/Main.java]: lol java 14 moment

File: src/main/java/xyz/whoneedspacee/ssmos/abilities/Ability.java (hunk 4 above)
Comment (Spyrxo) [reply src/main/java/xyz/whoneedspacee/ssmos/abilities/Ability.java:1]: idk it was just generic, having ssm ssmos ssm everyone seemed kinda a meme

File: src/main/java/xyz/whoneedspacee/ssmos/abilities/boss/WhirlpoolShotgun.java (hunk 5 above)
Comment (Spyrxo) [reply src/main/java/xyz/whoneedspacee/ssmos/abilities/boss/WhirlpoolShotgun.java:21]: intellij auto-refactor outcome...

File: src/main/java/xyz/whoneedspacee/ssmos/kits/Kit.java (hunk 6 above)
Comment (Spyrxo) [reply src/main/java/xyz/whoneedspacee/ssmos/kits/Kit.java]: intellij auto-refactor outcome...

File: src/main/java/xyz/whoneedspacee/ssmos/kits/Kit.java (hunk 7 above)
Comment (Spyrxo) [reply src/main/java/xyz/whoneedspacee/ssmos/kits/Kit.java]: intellij auto-refactor outcome...

File: src/main/java/xyz/whoneedspacee/ssmos/managers/disguises/Disguise.java (hunk 8 above)
Comment (Spyrxo) [reply src/main/java/xyz/whoneedspacee/ssmos/managers/disguises/Disguise.java]: tru that tru that. and yes it does. the funny when the game doesn't even have guardian sounds. 🧠

**Comment by Spyrxo** [comment]
unfortunate ssmos L . i appreciate ur quick read. pls consider looking more thorough later on, maven helps all the ide noobs out immensly.

**Comment by Whoneedspacee** [comment]
If you reduce the files changed I'd be open to looking into it, but this is just too much to ask when you didn't even review the files yourself. I would love for the project to be more accessible though so it is a shame.

**Comment by Spyrxo** [comment]
the file-size is probably due to the heirachy refactor, so that it actually works with maven and follows git standards. if you pull the code yourself and actually check it generally ull see its all exactly the same.

**Comment by Whoneedspacee** [comment]
I am bored so I am going to review this

File: src/main/java/xyz/whoneedspacee/ssmos/kits/ssmos/OSKitEnderman.java [hunk 9]
```diff
@@ -0,0 +1,12 @@
+package xyz.whoneedspacee.ssmos.kits.ssmos;
+
+import xyz.whoneedspacee.ssmos.kits.original.KitEnderman;
+
+public class OSKitEnderman extends KitEnderman {
+
+    public OSKitEnderman() {
+        super();
+        this.damage = 6.5;
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/kits/ssmos/OSKitEnderman.java:9]: Lol sure why not

File: src/main/java/xyz/whoneedspacee/ssmos/kits/ssmos/OSKitSkeletonHorse.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+        this.armor = 6;
+    }
+
+    @Override
+    public void initializeKit() {
+        super.initializeKit();
+        setArmorSlot(Material.CHAINMAIL_BOOTS, 0);
+        BoneRush rush = getAttributeByClass(BoneRush.class);
+        if(rush != null) {
+            rush.duration_ms = 1200;
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/kits/ssmos/OSKitSkeletonHorse.java:20]: Based

File: src/main/java/xyz/whoneedspacee/ssmos/kits/ssmos/OSKitSpider.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+        super();
+    }
+
+    @Override
+    public void initializeKit() {
+        super.initializeKit();
+        SpiderJump jump = getAttributeByClass(SpiderJump.class);
+        if(jump != null) {
+            jump.energy_to_jump = 0.1f;
+            jump.expUsed = 0.1f;
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/kits/ssmos/OSKitSpider.java:18]: Absolutely toxic

File: src/main/java/xyz/whoneedspacee/ssmos/managers/disguises/Disguise.java [hunk 12]
```diff
@@ -294,38 +294,88 @@ public boolean getShowAttackAnimation() {
...
+                sound = Sound.ENDERMAN_HIT;
+                break;
+            case GHAST:
+                sound = Sound.GHAST_SCREAM;
+                break;
+            case GIANT:
+            case ZOMBIE:
+                sound = Sound.ZOMBIE_HURT;
+                break;
+            case HORSE: // lmao virgin bukkit entity type doesn't differentiate between horse variants
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/managers/disguises/Disguise.java:335]: 🥇

File: src/main/java/xyz/whoneedspacee/ssmos/utilities/DamageUtil.java [hunk 13]
```diff
@@ -93,39 +94,103 @@ public static boolean isIntangible(LivingEntity check) {
...
+        if (type == EntityType.GUARDIAN) {
+            ((CraftWorld) damagee.getWorld()).getHandle().makeSound(
+                    damagee.getLocation().getX(),
+                    damagee.getLocation().getY(),
+                    damagee.getLocation().getZ(),
+                    "mob.guardian.hit", volume, pitch);
+            return;
+        }
+
+        // gholemFix: ik exactly where space copied the previous code from, so u can thank me later for making it better here
```
Comment (Whoneedspacee) [inline src/main/java/xyz/whoneedspacee/ssmos/utilities/DamageUtil.java:110]: Oh no all the poor playdamagesound editors

**Comment by Whoneedspacee** [comment]
Had to delete the template server, pretty sure we shouldn't be hosting server jars they need to be built for a reason?

If there's a good argument otherwise I'd revert that commit.

**Pull request merged**